src/options.rs

```diff
@@ -0,0 +1,118 @@
+// Copyright 2020 Nervos Core Dev
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::{ffi::CStr, path};
+
+use crate::{db_options::new_cache, ffi, ffi_util, ColumnFamilyDescriptor, Error, Options};
+
+#[derive(Clone)]
+pub struct FullOptions {
+    pub db_opts: Options,
+    pub cf_descriptors: Vec<ColumnFamilyDescriptor>,
+}
+
+impl FullOptions {
+    pub fn load_from_file<P>(
+        file: P,
+        cache_size: Option<usize>,
+        ignore_unknown_options: bool,
+    ) -> Result<Self, Error>
+    where
+        P: AsRef<path::Path>,
+    {
+        let cpath = ffi_util::to_cpath(
+            file,
+            "Failed to convert path to CString when load config file.",
+        )?;
+
+        let cache = if let Some(cache_size) = cache_size {
```

Comment:
Will Rust automatically free the cache?

Reply:
I'm afraid not.

But there is no way to avoid it in current APIs.
We already has the same issue if we use caches:
https://github.com/nervosnetwork/rust-rocksdb/blob/686cdff2e4946c1b3d396e07acdabcec782de2b2/src/db_options.rs#L37-L39
https://github.com/nervosnetwork/rust-rocksdb/blob/686cdff2e4946c1b3d396e07acdabcec782de2b2/src/db_options.rs#L307-L316

Reply:
Got it. This is the cache for RocksDB, not only for reading file options.